src/components/createRoomModal.vue

```diff
@@ -0,0 +1,255 @@
+<template>
+  <div class="modal-mask">
+    <div class="modal-wrapper">
+      <div class="modal-container">
+        <div>
+          <button id="close-button" @click="$emit('close')">
+            X
+          </button>
+        </div>
+        <br>
+        <div id="label-container">
+          Create Room
+        </div>
+
+        <div class="modal-body name-container">
+          <slot name="body">
+            <div id="name-label">
+              Room Name
+            </div>
+            <input v-model="roomNameInput" placeholder="Enter room name">
+          </slot>
+        </div>
+
+        <div class="modal-body type-container">
+          <slot name="body">
+            <div id="type-label">
+              Room Type
+            </div>
+            <select v-model="typeInput" id="type-select">
+              <option disabled value="">Select a type...</option>
+              <option value=true>Public</option>
+              <option value=false>Private</option>
+            </select>
+          </slot>
+        </div>
+
+        <div class="modal-body time-container">
+          <slot name="body">
+            <div v-bind:class="{'box-active': timeInput === 10, 'box-inactive': timeInput !== 10}" @click="putTime(10)">10 min</div>
+            <div v-bind:class="{'box-active': timeInput === 5, 'box-inactive': timeInput !== 5}" @click="putTime(5)">5 min</div>
+            <div v-bind:class="{'box-active': timeInput === 3, 'box-inactive': timeInput !== 3}" @click="putTime(3)">3 min</div>
+            <div v-bind:class="{'box-active': timeInput === 1, 'box-inactive': timeInput !== 1}" @click="putTime(1)">1 min</div>
+          </slot>
+        </div>
+
+        <div class="button-container">
+            <button :disabled="!valid" 
+                    v-bind:class="{'submit-button': valid, 'submit-disabled': !valid}"
+                    @click="createRoom">
+              {{valid ? 'Create!' : 'Missing Fields'}}
+            </button>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
```

Comment:
i suggest use b-modal instead of defining it from scratch. we have bootstrap-vue installed naman

Reply:
in the future for refactor c: migrating the styling to b-modal and adding prop referencing can take some time since its breaking a few stuff in my testing